operator/pkg/securedcluster/scanner/auto_sense.go

```diff
@@ -29,6 +46,26 @@ func AutoSenseLocalScannerSupport(ctx context.Context, client ctrlClient.Client,
 	return false, errors.Errorf("invalid spec.scanner.scannerComponent %q", scannerComponent)
 }
 
+func isRunningOnOpenShift(ctx context.Context, client ctrlClient.Client) (bool, error) {
+	log := ctrlLog.FromContext(ctx)
+
+	clusterVersion := &osconfigv1.ClusterVersion{}
+	key := ctrlClient.ObjectKey{Name: clusterVersionDefaultName}
+	err := client.Get(ctx, key, clusterVersion)
+	if err != nil && k8sErrors.IsNotFound(err) {
+		log.Info("Running on Kubernetes, OpenShift ClusterVersion was not found")
+		return false, err
+	} else if err != nil && meta.IsNoMatchError(err) {
+		log.Info(fmt.Sprintf("Running on Kubernetes, resource OpenShift ClusterVersion %q does not exist", clusterVersionDefaultName))
+		return false, nil
+	} else if err != nil {
+		log.Error(err, "Failed to get ClusterVersion")
+		return false, err
+	}
+
+	return clusterVersion.Spec.ClusterID != "", nil
```

Comment:
Instead checking the object's UUID? 🤔

Reply:
Why would we need to do it?

Reply:
This field is guarantied according to the [schema description](https://github.com/openshift/api/blob/c689b3938fab2ff48f7735ac410e7565d6e53426/config/v1/types_cluster_version.go#L41) 
I think you are right and it makes sense to just `return true`. Especially since it's clumsy to fetch some nested fields from unstructured object and if we were able to find `ClusterVersion` CR that should be sufficient